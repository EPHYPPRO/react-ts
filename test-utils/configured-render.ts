import React, { Props } from 'react';
import { render } from 'react-testing-library';

export function getConfiguredRender<P>(
  ui: React.ReactElement<any>,
  { ...options } = {}
) {
  return (props?: P & Props<any>) => {
    const utils = render(React.cloneElement(ui, props), options);
    return { ...utils, component: utils.container.firstChild };
  };
}
