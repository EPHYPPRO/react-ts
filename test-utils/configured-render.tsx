import React, { Props } from 'react';
import { render, RenderOptions } from 'react-testing-library';
import {
  ContextsProvider,
  ContextsOptions
} from './contexts-provider';
import { getConfiguredStore } from 'src/store';

export type Options = RenderOptions & ContextsOptions;

export function getConfiguredRender<P>(ui: React.ReactElement<any>) {
  return (props?: Partial<P & Props<any>>, options: Options = {}) => {
    let store = getConfiguredStore(options.storeInitialState);

    const utils = render(
      <ContextsProvider { ...options } store={ store }>
        { React.cloneElement(ui, props) }
      </ContextsProvider>,
      options
    );

    const rerender = (
      props?: P & Props<any>,
      contextsOptions?: ContextsOptions
    ) => {
      store = getConfiguredStore(
        contextsOptions
          ? contextsOptions.storeInitialState
          : options.storeInitialState
      );

      utils.rerender(
        <ContextsProvider
          { ...(contextsOptions ? contextsOptions : options) }
          store={ store }
        >
          { React.cloneElement(ui, props) }
        </ContextsProvider>
      );
    };

    const getStoreState = () => {
      return store.getState();
    };

    const newUtils = {
      ...utils,
      rerender,
      component: utils.container.firstChild as HTMLElement,
      getStoreState
    };

    return newUtils;
  };
}
