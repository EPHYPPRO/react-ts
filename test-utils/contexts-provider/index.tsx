import React, { FC } from 'react';
import { ReduxProvider } from './redux-provider';
import { State } from 'src/store/state';
import { Store } from 'redux';

export interface ContextsProps {
  store: Store<State>;
}

export interface ContextsOptions {
  storeInitialState?: State;
}

export const ContextsProvider: FC<ContextsProps> = ({
  children,
  store
}) => <ReduxProvider store={ store }>{ children }</ReduxProvider>;
