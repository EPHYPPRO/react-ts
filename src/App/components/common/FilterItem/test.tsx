import * as React from 'react';
import { FilterItem, FilterItemProps } from '.';
import { getConfiguredRender } from 'test-utils/configured-render';
import fake from 'casual';

const render = getConfiguredRender<FilterItemProps>(<FilterItem />);

describe('FilterItem', () => {
  it('should render properly', async () => {
    const { component } = render({
      label: 'label text',
      children: 'content'
    });

    expect(component).toMatchSnapshot();
  });

  it('should render children and do not render label element', async () => {
    const children = fake.text;
    const { container, queryByTestId } = render({ children });

    expect(container).toHaveTextContent(new RegExp(`${children}`));
    expect(queryByTestId('label')).toBeFalsy();
  });

  it('should render label', async () => {
    const label = fake.title;
    const { getByTestId } = render({ label });

    expect(getByTestId('label')).toHaveTextContent(
      new RegExp(label)
    );
  });
});
