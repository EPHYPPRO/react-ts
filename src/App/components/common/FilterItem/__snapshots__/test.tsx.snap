// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FilterItem should render properly 1`] = `
<div
  class="filter-item"
>
  <div
    class="label"
    data-testid="label"
  >
    label text
  </div>
  content
</div>
`;
